### PR TITLE
[v6r16] DynamicMonitoring Profiler

### PR DESCRIPTION
**Pull request by Sbalbp**
Added profiling utility used to retrieve dynamic information about a process. This is used by the dynamic monitoring system to get the information that will be stored into ElasticSearch.

### CODE REVIEW

File: Core/Utilities/Profiler.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      return S_ERROR( e )
+
+    return S_OK( result )
+
+  def memoryUsage( self ):
+    """
+    Returns the memory usage of the process in MB
+    """
+    try:
+      result = self.process.get_memory_info()[0] / float( 2 ** 20 )
```
Comment (chaen) [inline Core/Utilities/Profiler.py]: maybe a comment on what is this arithmetic ?

**Comment by chaen** [comment]
But how is it supposed to be used ?

File: Core/Utilities/Profiler.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Profiling class for updated information on process status
+"""
+
+__RCSID__ = "$Id$"
+
+import psutil
+import datetime
+from DIRAC import S_OK, S_ERROR
+
+class Profiler:
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: inherits from object?

File: Core/Utilities/Profiler.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  Calls to the different methods of the class will return the current state of the process.
+  """
+
+  def __init__( self, pid = None ):
+    """
+    :param str pid: PID of the process to be profiled
+    """
+    self.__setProcess( pid )
+
+  def __setProcess( self, pid = None ):
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: Is there a real need for this function? Why not directly in the __init__ ?

File: Core/Utilities/Profiler.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
+"""
+Profiling class for updated information on process status
+"""
+
+__RCSID__ = "$Id$"
+
+import psutil
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: Note: psutil is not part of the standard library: took note in https://github.com/DIRACGrid/Externals/issues/11

File: Core/Utilities/Profiler.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+  def __setProcess( self, pid = None ):
+    """
+    Set the current process
+    :param str pid: PID of the process to be profiled
+    """
+    if pid:
+      self.process = psutil.Process( int( pid ) )
+    else:
+      self.process = psutil.Process()
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: if pid is not given, the current process PID is used. Do you see a use case for that?

File: Core/Utilities/Profiler.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
+"""
+Profiling class for updated information on process status
+"""
+
+__RCSID__ = "$Id$"
+
+import psutil
+import datetime
+from DIRAC import S_OK, S_ERROR
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: There's not even one call to gLogger here... would you add some, please?

File: Core/Utilities/Profiler.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    return S_OK( self.process.pid )
+
+  def status( self ):
+    """
+    Returns the process status
+    """
+    try:
+      result = self.process.status()
+    except Exception as e:
+      return S_ERROR( e )
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: For all the constructs like this: would you be able to specify better the exception type, and log a proper error?

**Comment by zmathe** [comment]
it is recreated #2971

**Pull request closed without merging**